design/fips-compliance.md

```diff
@@ -0,0 +1,199 @@
+ `. For example, `golang:1.12` should turn into `goboring/golang:1.12.17b4`
+ - Enable cgo by setting `CGO_ENABLED=1` in the env
+ - Verify that the `authenticator` binary uses the BoringCrypto libraries by
+  running `go tool nm authenticator` and verifying that it has symbols named
+   *_Cfunc__goboringcrypto_*
+   
+More info on the required changes can be found [here](https://github.com/cyberark/conjur-authn-k8s-client/pull/97)
+
+### Non-FIPS compliant image
+
+We should decide if we still want to release a version of the `conjur-authn-k8s-client`
+that is not FIPS compliant (i.e that uses `golang` as the base image). Let's
+look into some pros and cons of both approaches.
+ 
+ | **Option**                             | **Implications**                                                                                                                                                                                                                                                                                                                                                        |
+ |----------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+ | Release only `goboring/golang` version | - Customers use the client and don't require FIPS compliancy, will now miss libraries that are available in `golang` but are not available in `goboring/golang`                                                                                                                                                                                                       |
+ | Release both versions                  | <ul><li>One more Dockerfile to maintain. Any change made to the original Dockerfile should be done also to `Dockerfile.fips` which can be easily forgotten</li><li>Another version to test. Our UTs will stay the same but we will need to run integration tests on both images, in order to release them. This adds complexity to our tests and doubles the tests run-time</li></ul> |
+
+From the table above, I think that we should release only the FIPS compliant
+version. The advantages that customers will get by using the `golang` image
+as the base image is not worth the maintenance burden that Releasing both
+versions will bring to the development team.
+
+### Red Hat image
+
+We will also release a FIPS compliant Red Hat image of the `conjur-authn-k8s
+-client` to our Red Hat registry. Before we do that, we should create a 
+`conjur-authn-k8s-client` FIPS project in the RH container registry.
+
+TBD - how do we create the FIPS project in the RH container registry?
```

Comment:
In the [implementation PR](https://github.com/cyberark/conjur-authn-k8s-client/pull/97#discussion_r402479843) @izgeri says: 
```
we can't push the RH FIPS image without creating an authn-k8s FIPS project in the RH container registry - so that should maybe get pushed to a follow-up ticket. We can leave the build steps in this PR, but I don't think we can publish that image yet in this PR
```

this is what i meant should be done. @izgeri can you elaborate please on how this can be done?

Reply:
It depends how we want to do this. We have a project for authn-k8s already - so we can push a new image to that project with a different tag (e.g. `0.17.0-fips`) and that won't require a separate project. That's probably the most sensible way to proceed - otherwise, we'd create a separate `authn-k8s-fips` project.

Reply:
so if we release only the FIPS compliant version (i.e replace the current one with it) then we don't need to do anything in RH?